lib/cext/preprocess.rb

```diff
@@ -114,12 +114,25 @@ def self.patch(file, contents, directory)
     contents
   end
 
+  require 'stringio'
+
   if __FILE__ == $0
-    puts "#line 1 \"#{ARGF.filename}\""
+    out = StringIO.new
 
-    contents = patch(ARGF.filename, File.read(ARGF.filename), Dir.pwd)
+    original = File.read(ARGF.filename)
```

Comment:
Can you extract `ARGF.filename` in a variable `file` since it's used in many places now?